test-case/check-alsabat.sh

```diff
@@ -64,11 +72,52 @@ function __upload_wav_file
     done
 }
 
+function set_pga_to_unity_gain
+{
+    tmp=$(amixer controls | grep "$pga" | grep Volume)
+    search="name="
+    cname=${tmp#*$search}
+
+    # Get volume min and step to compute value for cset
+    # for 0 dB gain. The amixer line looks like
+    # "| dBscale-min=-50.00dB,step=1.00dB,mute=1"
+    scale=$(amixer cget name="$cname" | grep "dBscale")
+    search="dBscale-min="
+    tmp=${scale#*$search}
+    min_db="${tmp%%dB*}"
+    search="step="
+    tmp=${scale#*$search}
+    step_db="${tmp%%dB*}"
+
+    # Get multiplied by 100 values by removing decimal dot
+    min_x100="${min_db//.}"
+    step_x100="${step_db//.}"
+    val=$(printf %d "$(((-$min_x100) / $step_x100))")
+
+    # Apply the computed 0 dB gain
+    amixer cset name="$cname" "$val"
+}
+
 # check the PCMs before alsabat test
 dlogi "check the PCMs before alsabat test"
 [[ $(aplay -Dplug$pcm_p -d 1 /dev/zero -q) ]] && die "Failed to play on PCM: $pcm_p"
 [[ $(arecord -Dplug$pcm_c -d 1 /dev/null -q) ]] && die "Failed to capture on PCM: $pcm_c"
 
+# Set PGAs for PCMs to 0 dB value
+test -n "$tplg" || die "Use -t or set environment variable TPLG to current topology"
```

Comment:
Please double-check but I suspect `func_opt_parse_option` gives something like that for free (it should)